### PR TITLE
Add resource requirements config to cluster installer

### PR DESCRIPTION
**Pull request by nacardin**
Resolves https://github.com/infinyon/fluvio/issues/543

### CODE REVIEW

File: src/cluster/src/start/k8.rs [hunk 1]
```diff
@@ -816,26 +930,86 @@ impl ClusterInstaller {
...
+                Cow::Borrowed(authorization_config_map),
+            ));
+        }
+
+        if let Some(ref resource_requirments) = &self.config.resource_requirments {
+            if let Some(ref sc_resource_requirments) = resource_requirments.sc {
+                if let Some(ref requests) = sc_resource_requirments.requests {
+                    if let Some(ref cpu) = requests.cpu {
+                        install_settings
+                            .push(("scResources.requests.cpu", Cow::Owned(cpu.to_string())));
```
Comment (sehz) [inline src/cluster/src/start/k8.rs:961]: can you make these as constant?

File: src/cluster/src/start/k8.rs [hunk 2]
```diff
@@ -816,26 +930,86 @@ impl ClusterInstaller {
...
+                    }
+                    if let Some(ref memory) = requests.memory {
+                        install_settings.push((
+                            "scResources.requests.memory",
```
Comment (sehz) [inline src/cluster/src/start/k8.rs:965]: make this as constant

File: src/cluster/src/start/k8.rs [hunk 3]
```diff
@@ -816,26 +930,86 @@ impl ClusterInstaller {
...
+                            .push(("spuResources.limits.memory", Cow::Owned(memory.to_string())));
+                    }
+                }
+            }
         }
 
+        use std::borrow::Borrow;
+        let install_settings = install_settings
+            .iter()
+            .map(|(k, v)| (*k, v.borrow()))
```
Comment (sehz) [inline src/cluster/src/start/k8.rs:1010]: pretty fancy trick!

**Review by sehz — CHANGES_REQUESTED** [review]
minor nitpicks

File: src/cluster/src/start/k8.rs (hunk 1 above)
Comment (nacardin) [reply src/cluster/src/start/k8.rs]: yes

File: src/cluster/src/start/k8.rs (hunk 2 above)
Comment (nacardin) [reply src/cluster/src/start/k8.rs]: yes

**Review by sehz — APPROVED** [review]

**Pull request merged**